api/v1alpha1/hostedcluster_types.go

```diff
@@ -478,6 +478,10 @@ type APIServerNetworking struct {
 	// pods using host networking cannot listen on this port. If not specified,
 	// 6443 is used.
 	Port *int32 `json:"port,omitempty"`
+
+	// AllowedCIDRBlocks is an allow list of CIDR blocks that can access the APIServer
+	// If not specified, traffic is allowed from all addresses.
+	AllowedCIDRBlocks []string `json:"allowedCIDRBlocks,omitempty"`
```

Comment:
Can we re-use whatever validation is on the upstream service object here?

Reply:
There seems to be some prior art wrt regex to match a cidr, like for example https://blog.markhatton.co.uk/2011/03/15/regular-expressions-for-ip-addresses-cidr-ranges-and-hostnames/

The reason I care is that making validation more lenient is never a problem, but making a too lenient validation (or none as right now) more strict is often a bigger deal, since it is a breaking change. So if possible I'd prefer to have something here.

Reply:
But is that a reason to continue like that?